File: Web/src/api/userGroupContainer.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetGroupContainerListByUser() {
  return apiClient.post("/userGroupContainer/GetGroupContainerListByUser").then(handleResponse).catch(handleError)
}

function CreateGroupContainerByUser(data) {
  return apiClient.post("/userGroupContainer/CreateGroupContainerByUser", data).then(handleResponse).catch(handleError)
}

function DestroyGroupContainerByUser(data) {
  return apiClient.post("/userGroupContainer/DestroyGroupContainerByUser", data).then(handleResponse).catch(handleError)
}

function DelayGroupContainerByUser(data) {
  return apiClient.post("/userGroupContainer/DelayGroupContainerByUser", data).then(handleResponse).catch(handleError)
}

const userGroupContainerApi = {
  GetGroupContainerListByUser,
  CreateGroupContainerByUser,
  DestroyGroupContainerByUser,
  DelayGroupContainerByUser
}

export default userGroupContainerApi

File: Server/route/userGroupContainer.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserGroupContainerRoute(r *gin.Engine) {
	userContainer := r.Group("/userGroupContainer")
	userContainer.Use(middleware.JWTAuthMiddleware())
	{
		userContainer.POST("/GetGroupContainerListByUser", api.GetGroupContainerListByUser)
		userContainer.POST("/CreateGroupContainerByUser", api.CreateGroupContainerByUser)
		userContainer.POST("/DestroyGroupContainerByUser", api.DestroyGroupContainerByUser)
		userContainer.POST("/DelayGroupContainerByUser", api.DelayGroupContainerByUser)
	}
}
